src/sentry/workflow_engine/actions/notification_action/logic.py

```diff
@@ -0,0 +1,137 @@
+    rule_data = dict(action.data)
+
+    # Add the rule registry id to the rule data
+    rule_data["id"] = ACTION_TYPE_2_RULE_REGISTRY_ID[action.type]
+
+    # Add the action uuid to the rule data
+    rule_data["uuid"] = action.id
+
+    # Add the integration id to the rule data
+    integration_id_key = ACTION_TYPE_2_INTEGRATION_ID_KEY[action.type]
+    rule_data[integration_id_key] = action.integration_id
+
+    # Add the target identifier to the rule data if it exists
+    target_identifier_key = ACTION_TYPE_2_TARGET_IDENTIFIER_KEY.get(action.type)
+    if target_identifier_key:
+        rule_data[target_identifier_key] = action.target_identifier
+
+    # Add the target display to the rule data if it exists
+    target_display_key = ACTION_TYPE_2_TARGET_DISPLAY_KEY.get(action.type)
+    if target_display_key:
+        rule_data[target_display_key] = action.target_display
+
+    # Because the rule expects "actions" to be a list, we need to return a list with a single dict
+    return [rule_data]
+
+
+def form_issue_alert_models(
+    action: Action, detector: Detector, group_event: GroupEvent
+) -> tuple[Rule, RuleFireHistory, str]:
+    """
+    Builds the Rule and RuleFireHistory models from the Action model.
+    These models are used to trigger an issue alert notification.
+
+    :param action: Action model instance
+    :param detector: Detector model instance
+    :param group_event: GroupEvent model instance
+    :return: tuple of Rule, RuleFireHistory, and notification_uuid
+    """
+
+    # Create a notification uuid
+    notification_uuid = str(uuid.uuid4())
+
+    assert detector.project == group_event.project
+    # TODO(iamrajjoshi): Change the above assert to a check
+
+    # Build the Rule.data.actions json blob
+    rule_data_json_blob = build_rule_data_blob(action)
+
+    rule = Rule(
+        id=detector.id,
+        project=detector.project,
+        label=detector.name,
+        data={
+            "actions": rule_data_json_blob,
+        },
+        status=ObjectStatus.ACTIVE,
+        source=RuleSource.ISSUE,
+    )
+
+    rule_fire_history = RuleFireHistory(
+        project=detector.project,
+        rule=rule,
+        event_id=group_event.event_id,
+        group_id=group_event.group_id,
+        notification_uuid=notification_uuid,
+    )
+
+    rule.save()
+    rule_fire_history.save()
+
+    return rule, rule_fire_history, notification_uuid
+
+
+def invoke_issue_alert_registry(action: Action, detector: Detector, group_event: GroupEvent):
+    """
+    Invokes the issue alert registry and sends a notification.
+
+    :param action: Action model instance
+    :param detector: Detector model instance
+    :param group_event: GroupEvent model instance
+    """
+
+    rule, rule_fire_history, notification_uuid = form_issue_alert_models(
+        action, detector, group_event
+    )
+
+    # TODO(iamrajjoshi): Add a check to see if the rule has only one action
+    assert len(rule.data.get("actions", [])) == 1
```

Comment:
a little nervous that this will really increase the number of rules we have in the db; it seems like for each notification we're sending; we'll be creating a new rule (so if the workflow has 5 actions, we'll create 5 rules with 1 action each). It also seems like we're creating it every time, but not removing it and not reusing it.